Purpose: Get insserv building on kfreebsd, missing the argument values for posix_fadvise().
Fixes:   #?
Status:  Not yet submitted upstream.
---
Index: insserv/insserv.c
===================================================================
--- insserv.orig/insserv.c	2009-09-26 22:35:39.000000000 +0200
+++ insserv/insserv.c	2009-09-26 22:35:40.000000000 +0200
@@ -40,6 +40,13 @@
 #endif /* USE_RPMLIB */
 #include "listing.h"
 
+#if defined _XOPEN_SOURCE && (_XOPEN_SOURCE - 0) >= 600
+/* kfreebsd fail to provide working posix_fadvise */
+#  ifndef POSIX_FADV_SEQUENTIAL
+#    define posix_fadvise(fd, off, len, adv)        (-1)
+#  endif
+#endif
+
 #ifdef SUSE
 # define DEFAULT_START_LVL	"3 5"
 # define DEFAULT_STOP_LVL	"3 5"
